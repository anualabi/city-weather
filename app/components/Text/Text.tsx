import i18n from "i18n-js";
import { forwardRef } from "react";
import {
  StyleProp,
  StyleSheet,
  Text as RNText,
  TextProps as RNTextProps,
  TextStyle,
} from "react-native";

export interface TextProps extends RNTextProps {
  text?: string;
  txOptions?: i18n.TranslateOptions;
  style?: StyleProp<TextStyle>;
  children?: React.ReactNode;
}

const Text = forwardRef<RNText, TextProps>(function Text(props, ref) {
  const { txOptions, text, children, style, ...rest } = props;

  return (
    <RNText ref={ref} {...rest} style={[styles.text, style]}>
      {children}
    </RNText>
  );
});

const styles = StyleSheet.create({
  text: { fontSize: 16 },
});

export default Text;
