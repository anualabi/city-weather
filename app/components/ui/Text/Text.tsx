import i18n from "i18n-js";
import { forwardRef } from "react";
import {
  Text as RNText,
  TextProps as RNTextProps,
  StyleProp,
  StyleSheet,
  TextStyle,
} from "react-native";

import { translate, TxKeyPath } from "@/i18n";
import { colors } from "@/theme";

export interface TextProps extends RNTextProps {
  tx?: TxKeyPath;
  text?: string;
  txOptions?: i18n.TranslateOptions;
  style?: StyleProp<TextStyle>;
  children?: React.ReactNode;
}

const Text = forwardRef<RNText, TextProps>(function Text(props, ref) {
  const { tx, txOptions, text, children, style, ...rest } = props;

  const i18nText = tx && translate(tx, txOptions);
  const content = i18nText ?? text ?? children;

  return (
    <RNText ref={ref} {...rest} style={[styles.text, style]}>
      {content}
    </RNText>
  );
});

const styles = StyleSheet.create({
  text: { color: colors.text, fontSize: 16 },
});

export default Text;
