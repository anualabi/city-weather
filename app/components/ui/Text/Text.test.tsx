import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";

import Text, { TextProps } from "./Text";

describe("Text component", () => {
  test("should render correctly", () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render text content from the "text" prop', () => {
    const { getByText } = render(<Text text="Hello, World!" />);

    expect(getByText("Hello, World!")).toBeTruthy();
  });

  test("should render children when provided", () => {
    const { getByText } = render(<Text>Hello, World!</Text>);

    expect(getByText("Hello, World!")).toBeTruthy();
  });

  test("should apply custom styles when provided", () => {
    const customStyle: TextProps["style"] = { fontSize: 25, color: "red" };
    const { getByText } = render(
      <Text style={customStyle} text="Styled Text" />
    );
    expect(getByText("Styled Text")).toHaveStyle(customStyle);
  });
});
