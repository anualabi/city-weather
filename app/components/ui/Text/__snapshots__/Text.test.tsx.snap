// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#343434",
        "fontSize": 16,
      },
      undefined,
    ]
  }
/>
`;
